### ISSUE TITLE
`telepresence intercept` `panic`s

### ISSUE THREAD

**Issue by pmalek** [telepresenceio/telepresence#3405]
**Describe the bug**

```
telepresence intercept gw-kong --port kong-admin
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x102bf4e1a]

goroutine 1 [running, locked to thread]:
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*state).CreateRequest(0xc00083e9f0, {0x10379eee8, 0xc000858450})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/state.go:88 +0x3a
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.(*state).create(0xc0007fd740, {0x10379eee8, 0xc000858450})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/intercept/state.go:208 +0x22d
github.com/telepresenceio/telepresence/v2/pkg/client.WithEnsuredState({0x10379eee8, 0xc000858450}, 0x102c400a0?, 0x0, 0x0)
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/ensured_state.go:17 +0x4a
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.(*state).Run(0xc0007fd740, {0x10379eee8?, 0xc000823e00?})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/intercept/state.go:183 +0x205
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*state).Run(0xc00083e9f0, {0x10379eee8, 0xc000823e00})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/state.go:43 +0x45
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*Command).Run(0xc00059e780, 0xc0006c6600, {0xc0006a4240, 0x1, 0x3})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/intercept.go:238 +0x4c3
github.com/spf13/cobra.(*Command).execute(0xc0006c6600, {0xc0006a41e0, 0x3, 0x3})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000470000)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/telepresenceio/telepresence/v2/pkg/client/cli/output.Execute(0x10379eee8?)
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/output/output.go:114 +0x4b
github.com/telepresenceio/telepresence/v2/pkg/client/cli.Main({0x10379eee8, 0xc0006a43c0})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/main.go:73 +0x2ef
main.main()
        github.com/datawire/telepresence-pro/v2/cmd/telepresence/main.go:11 +0x2a
```

Even though the deployment has the port that I want:

```
$ k get deployments.apps -n kong gw-kong -o yaml | yq '.spec.template.spec.containers[0].ports'
- containerPort: 8001
  name: admin
  protocol: TCP
- containerPort: 8444
  name: admin-tls
  protocol: TCP
- containerPort: 8000
  name: proxy
  protocol: TCP
- containerPort: 8443
  name: proxy-tls
  protocol: TCP
- containerPort: 8100
  name: status
  protocol: TCP
$ telepresence intercept gw-kong --service gw-kong-admin --port=8001:admin
telepresence intercept: error: Deployment gw-kong.kong has no interceptable port matching service gw-kong-admin, port admin
$ telepresence intercept gw-kong --service gw-kong-admin --port=8001
telepresence intercept: error: Deployment gw-kong.kong has multiple interceptable ports in service gw-kong-admin.
Please specify the port you want to intercept by passing the --port=<local:svcPortName> flag.
$ telepresence intercept gw-kong --service gw-kong-admin --port=admin
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x102800e1a]

goroutine 1 [running, locked to thread]:
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*state).CreateRequest(0xc0004cd3c8, {0x1033aaee8, 0xc00080d050})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/state.go:88 +0x3a
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.(*state).create(0xc0007fe960, {0x1033aaee8, 0xc00080d050})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/intercept/state.go:208 +0x22d
github.com/telepresenceio/telepresence/v2/pkg/client.WithEnsuredState({0x1033aaee8, 0xc00080d050}, 0x10284c0a0?, 0x0, 0x0)
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/ensured_state.go:17 +0x4a
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.(*state).Run(0xc0007fe960, {0x1033aaee8?, 0xc00080cab0?})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/intercept/state.go:183 +0x205
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*state).Run(0xc0004cd3c8, {0x1033aaee8, 0xc00080cab0})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/state.go:43 +0x45
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*Command).Run(0xc0007c4780, 0xc000806600, {0xc0006251c0, 0x1, 0x4})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/intercept.go:238 +0x4c3
github.com/spf13/cobra.(*Command).execute(0xc000806600, {0xc000625180, 0x4, 0x4})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000807800)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/telepresenceio/telepresence/v2/pkg/client/cli/output.Execute(0x1033aaee8?)
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/output/output.go:114 +0x4b
github.com/telepresenceio/telepresence/v2/pkg/client/cli.Main({0x1033aaee8, 0xc0007f63c0})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/main.go:73 +0x2ef
main.main()
        github.com/datawire/telepresence-pro/v2/cmd/telepresence/main.go:11 +0x2a
```

```
telepresence status
User Daemon: Running
  Version           : v2.16.1
  Executable        : /opt/homebrew/bin/telepresence
  Install ID        : XXXXXXXXX
  Status            : Connected
  Kubernetes server : https://127.0.0.1:63155
  Kubernetes context: XXXXXXXXX
  Connection name   : XXXXXXXXX
  Namespace         : XXXXXXXXX
  Manager namespace : ambassador
  Intercepts        : 0 total
Root Daemon: Running
  Version    : v2.16.1
  Version    : v2.16.1
  DNS        :
    Remote IP       : 127.0.0.1
    Exclude suffixes: [.com .io .net .org .ru]
    Include suffixes: []
    Timeout         : 8s
  Also Proxy : (0 subnets)
  Never Proxy: (1 subnets)
    - 127.0.0.1/32
Ambassador Cloud:
  Status      : Logged in
  User ID     : XXXXXXXXX
  Account ID  : XXXXXXXXX
  User Name   : XXXXXXXXX
  Email       : XXXXXXXXX
  Account Name: XXXXXXXXX
  Plan        : Lite
Traffic Manager: Connected
  Version : v2.16.1
Intercept Spec:
```

---

The above does work when I use:

```
telepresence intercept gw-kong --service gw-kong-admin --port=8001:8001
Using Deployment gw-kong
   Intercept name         : gw-kong
   State                  : ACTIVE
   Workload kind          : Deployment
   Destination            : 127.0.0.1:8001
   Service Port Identifier: kong-admin
   Volume Mount Error     : sshfs is not installed on your local machine
   Intercepting           : all TCP requests
```

---

**Versions (please complete the following information):**

 - Output of `telepresence version`

    ```
    telepresence version
    Client         : v2.16.1
    Root Daemon    : v2.16.1
    User Daemon    : v2.16.1
    Traffic Manager: v2.16.1
    ```

 - Operating system of workstation running `telepresence` commands

   Mac OS, 13.6

 - Kubernetes environment and Version
  
   kind 1.28.0

**Comment by cindymullins-dw** [comment]
Hi @pmalek , is the Service you’re trying to access using numeric targetPorts?  If so, it’s recommended that you use a symbolic name for the target port rather than numeric.  A numeric port number forces Telepresence to inject an init-container that uses iptables to remap the port.  So in certain cases that can cause errors.

**Comment by pmalek** [comment]
The ports on the `Service` are as follows:

```
  ports:
  - name: kong-proxy
    nodePort: 31200
    port: 80
    protocol: TCP
    targetPort: 8000
  - name: kong-proxy-tls
    nodePort: 30371
    port: 443
    protocol: TCP
    targetPort: 8443
```

The `Deployment` that I use as workload here is called `gw-kong`.

The following works:

```
telepresence intercept --port 8080:80 gw-kong
```

The following panics:

```
telepresence intercept --port kong-proxy gw-kong
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x10474ee1a]

goroutine 1 [running, locked to thread]:
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*state).CreateRequest(0xc000013830, {0x1052f8ee8, 0xc000a3b110})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/state.go:88 +0x3a
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.(*state).create(0xc000a3e5a0, {0x1052f8ee8, 0xc000a3b110})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/intercept/state.go:208 +0x22d
github.com/telepresenceio/telepresence/v2/pkg/client.WithEnsuredState({0x1052f8ee8, 0xc000a3b110}, 0x10479a0a0?, 0x0, 0x0)
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/ensured_state.go:17 +0x4a
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.(*state).Run(0xc000a3e5a0, {0x1052f8ee8?, 0xc000a3ab70?})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/intercept/state.go:183 +0x205
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*state).Run(0xc000013830, {0x1052f8ee8, 0xc000a3ab70})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/state.go:43 +0x45
github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept.(*Command).Run(0xc0002a9900, 0xc00013e900, {0xc0005107e0, 0x1, 0x3})
        github.com/datawire/telepresence-pro/v2/pkg/client/cli/intercept/intercept.go:238 +0x4c3
github.com/spf13/cobra.(*Command).execute(0xc00013e900, {0xc000510780, 0x3, 0x3})
        github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00013fb00)
        github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/telepresenceio/telepresence/v2/pkg/client/cli/output.Execute(0x1052f8ee8?)
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/output/output.go:114 +0x4b
github.com/telepresenceio/telepresence/v2/pkg/client/cli.Main({0x1052f8ee8, 0xc00038b830})
        github.com/telepresenceio/telepresence/v2@v2.16.1/pkg/client/cli/main.go:73 +0x2ef
main.main()
        github.com/datawire/telepresence-pro/v2/cmd/telepresence/main.go:11 +0x2a
```

Errors (especially those that suggest user what to do) are fine. `panic`s are not.

**Comment by cindymullins-dw** [comment]
Ok, thanks for confirming.

**Comment by thallgren** [comment]
Fixed in 2.17.1
